src/components/TranslateDialog.vue

```diff
@@ -0,0 +1,194 @@
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<NcModal @close="$emit('close')">
+		<div class="translate-dialog">
+			<h2> {{ t('spreed', 'Translate message') }} </h2>
+			<label for="from">
+				{{ t('spreed', 'From') }}
+			</label>
+			<NcSelect v-model="selectedFrom"
+				input-id="from"
+				:aria-label="t('spreed', 'From:')"
+				:placeholder="t('spreed', 'From:')"
+				:options="availableFroms"
+				:clearable="false"
+				track-by="from"
+				no-wrap
+				@input="handleSelectFrom" />
+			<label for="to">
+				{{ t('spreed', 'To') }}
+			</label>
+			<NcSelect v-model="selectedTo"
+				input-id="to"
+				:aria-label="t('spreed', 'To:')"
+				:placeholder="t('spreed', 'To:')"
+				:options="availableTos"
+				:clearable="false"
+				track-by="To"
+				no-wrap />
+			<p class="translate-dialog__message translate-dialog__message-source">
+				{{ text }}
+			</p>
+			<p class="translate-dialog__message translate-dialog__message-translation">
+				{{ translatedMessage }}
+			</p>
+			<div class="translate-dialog__buttons">
+				<NcButton :disabled="loading"
+					class="translate-dialog__button"
+					@click="translateMessage">
+					<template v-if="loading" #icon>
+						<NcLoadingIcon />
+					</template>
+					{{ t('spreed', 'Translate') }}
+				</NcButton>
+			</div>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import { showError } from '@nextcloud/dialogs'
+
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+import NcSelect from '@nextcloud/vue/dist/Components/NcSelect.js'
+
+import { translateText } from '../services/messagesService.js'
+
+export default {
+	name: 'TranslateDialog',
+
+	components: {
+		NcModal,
+		NcSelect,
+		NcButton,
+		NcLoadingIcon,
+	},
+
+	props: {
+		availableTranslations: {
+			type: Array,
+			default: () => [],
+		},
+
+		text: {
+			type: String,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			translatedMessage: '',
+			availableTos: [],
+			selectedFrom: '',
+			selectedTo: '',
+			loading: false,
+		}
+	},
+
+	computed: {
+		availableFroms() {
+			return Array.from(new Set(this.availableTranslations.map(element => element.from)))
+		},
+	},
```

Comment:
```js
		optionsFrom() {
			return this.availableFroms.map(element => {
				const model = this.availableTranslations.find(language => language.from === element)
				return {
					id: model.from,
					label: model.fromLabel,
				}
			})
		},

		optionsTo() {
			return this.availableTranslations.filter(language => language.from === this.selectedFrom?.id)
				.map(model => {
					return {
						id: model.to,
						label: model.toLabel,
					}
				})
		},
```

Something like this could be used for Select options